cmd/nerdctl/container_cp_acid_linux_test.go

```diff
@@ -0,0 +1,180 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"gotest.tools/v3/assert"
+	"gotest.tools/v3/icmd"
+
+	"github.com/containerd/nerdctl/v2/pkg/containerutil"
+	"github.com/containerd/nerdctl/v2/pkg/rootlessutil"
+	"github.com/containerd/nerdctl/v2/pkg/testutil"
+)
+
+// This is a separate set of tests for cp specifically meant to test corner or extreme cases that do not fit in the normal testing rig
+// because of their complexity
+
+func TestAcidCopy(t *testing.T) {
```

Comment:
Does Acid here mean https://en.wikipedia.org/wiki/ACID  ([atomicity](https://en.wikipedia.org/wiki/Atomicity_(database_systems)), [consistency](https://en.wikipedia.org/wiki/Consistency_(database_systems)), [isolation](https://en.wikipedia.org/wiki/Isolation_(database_systems)), [durability](https://en.wikipedia.org/wiki/Durability_(database_systems)))?

Reply:
Probably tests should have the `TestCopy...` prefix

Reply:
"Acid test" generally would be a really hard to pass, crucial test (https://www.merriam-webster.com/dictionary/acid%20test)

It is also a nod to the old browsers "acid tests" (https://en.wikipedia.org/wiki/Acid1 https://en.wikipedia.org/wiki/Acid2 https://en.wikipedia.org/wiki/Acid3)

I can rename to TestCopyAcid to facilitate `-run TestCopy*`.

Reply:
The overall point is that there is two tiers of tests - "the happy path" kind of tests, and the "acid" kind.

"Acid" has very few tests, will very likely break on you very fast, and you can use it instead of the full test-suite to quickly **validate** if something is broken or not - but then it is probably really hard to debug if it fails.

"Happy path" should have many test, not very challenging, easy to debug, will run much longer, and is more useful for incremental debugging.